packages/botonic-react/src/webchat/webchat-input-panel/index.tsx

```diff
@@ -0,0 +1,116 @@
+import { getFullMimeWhitelist } from '../../message-utils'
+import { Attachment } from '../components/attachment'
+import { EmojiPicker, OpenedEmojiPicker } from '../components/emoji-picker'
+import { PersistentMenu } from '../components/persistent-menu'
+import { SendButton } from '../components/send-button'
+import { DeviceAdapter } from '../devices/device-adapter'
+import { Textarea } from '../textarea'
+import { UserInputContainer } from './styles'
+
+interface WebchatInputPanelProps {
+  persistentMenu: any
+  enableEmojiPicker: boolean
+  enableAttachments: boolean
+  handleAttachment: (event: any) => void
+  textareaRef: React.MutableRefObject<HTMLTextAreaElement>
+  deviceAdapter: DeviceAdapter
+  onUserInput?: (event: any) => Promise<void>
+}
+
+export const WebchatInputPanel = ({
+  persistentMenu,
+  enableEmojiPicker,
+  enableAttachments,
+  handleAttachment,
+  textareaRef,
+  deviceAdapter,
+  onUserInput,
+}: WebchatInputPanelProps) => {
+  const {
+    getThemeProperty,
+    sendText,
+    togglePersistentMenu,
+    toggleEmojiPicker,
+    webchatState,
+  } = useContext(WebchatContext)
+
+  const handleSelectedEmoji = event => {
+    textareaRef.current.value += event.emoji
+    textareaRef.current.focus()
+  }
+
+  const handleEmojiClick = () => {
+    toggleEmojiPicker(!webchatState.isEmojiPickerOpen)
+  }
+
+  const handleMenu = () => {
+    togglePersistentMenu(!webchatState.isPersistentMenuOpen)
+  }
+
+  const sendTextAreaText = async () => {
+    await sendText(textareaRef.current.value)
+    textareaRef.current.value = ''
+  }
+
+  const sendChatEvent = async chatEvent => {
+    const chatEventInput = {
+      id: uuidv4(),
+      type: INPUT.CHAT_EVENT,
+      data: chatEvent,
+    }
+    if (onUserInput) {
+      onUserInput({
+        user: webchatState.session.user,
+        input: chatEventInput,
+        session: webchatState.session,
+        lastRoutePath: webchatState.lastRoutePath,
+      })
+    }
+  }
+
+  return (
+    <UserInputContainer
+      style={{
+        ...getThemeProperty(WEBCHAT.CUSTOM_PROPERTIES.userInputStyle),
+      }}
+      className='user-input-container'
+    >
+      {webchatState.isEmojiPickerOpen && (
+        <OpenedEmojiPicker
+          // height={webchatState.theme.style.height} // Revisar si es necessari
+          onEmojiClick={handleSelectedEmoji}
+          onClick={handleEmojiClick}
+        />
+      )}
+
+      <PersistentMenu onClick={handleMenu} persistentMenu={persistentMenu} />
+
+      <Textarea
+        deviceAdapter={deviceAdapter}
+        persistentMenu={persistentMenu}
+        textareaRef={textareaRef}
+        sendChatEvent={sendChatEvent}
+        sendTextAreaText={sendTextAreaText}
+      />
+
+      <EmojiPicker
+        enableEmojiPicker={enableEmojiPicker}
+        onClick={handleEmojiClick}
+      />
+
+      <Attachment
+        enableAttachments={enableAttachments}
+        onChange={handleAttachment}
+        accept={getFullMimeWhitelist().join(',')}
```

Comment:
not for this PR, but I'm not a fan of having a join here, I think we can think of a more nicer approach